
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [260]:
%matplotlib inline
import pandas as pd
import numpy as np
from scipy import stats 
import matplotlib.pyplot as plt


# some nicer colors from http://colorbrewer2.org/
COLOR1 = '#7fc97f'
COLOR2 = '#beaed4'
COLOR3 = '#fdc086'
COLOR4 = '#ffff99'
COLOR5 = '#386cb0'

In [261]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')


In [262]:
# number of callbacks for balck-sounding names
bc=sum(data[data.race=='b'].call)


In [263]:
# number of callbacks for balck-sounding names
wc=sum(data[data.race=='w'].call)

call_data=np.array([bc,wc])

# total number of calls
N=wc+bc

call_data

array([ 157.,  235.])

In [278]:
# sample mean proportion of calls for black-sounding names
p=bc/N

#Rule of thumb test for CLT applicability: Np AND Nq >10

Np=p*N

Nq=N * (1-p)


print("N * p = ", Np, '\n' "N(1 - p) = ", Nq)

N * p =  157.0 
N(1 - p) =  235.0


## Q1 Response: 

The nature of callback data is binary. 

The problem can be solved through the statistical analysis of proportions. The distribution of sample proportion means are approximately normal for large sample sizes, where the mean value is not close to the extreme readings of 0 or 1. Normal distribution can be used to approximate the population mean distribution if both N x p and N(1 - p) are greater than 10 [where N=sample size and p=proportion mean]. 

In our case both N x p and  N (1 - p) are well above 10, which means we can use CLT to approximate sampling distribution of p.  


## Q2 Response:

We want to analyze whether the proportion of calls for black-sounding names is due to chance. In the context of this exercise We can assign 1 for every call a black-sounding name and zero for 


If black-sounding names had the same (or greater) chance of being called as white-sounding names, their proportion would be close to or above 0.5. This is our null hypothesis. Alternative hypothesis will be that proportion calls for black-sounding names is less than 0.5. 

H0: p * N >= 0.5 or (1-p)

H1: p * N < 0.5 or (1-p)




In [296]:
#fig, ax = plt.subplots(1, 1)

# Significance level: 0.05
# Type of hypothesis test: one-tailed
# Confidence Interval: 95 %

#Analysis Steps
# 1. calculate sample statistics:

pp = (p * Np + (1-p) * Nq) / (Np + Nq) # pooled probability

SE = round(np.sqrt(p * ( 1 - p ) * ( (1/Np) + (1/Nq) )), 3) # standard error


#2. Calculate Z statistic, p value and 95% confidence interval

z = (p - (1-p)) / SE

p_value = round(stats.norm.sf(abs(z)), 4)

r = binom.rvs(N, p, size=1000)

CI=np.percentile(r, [0, 95])

print('Z-score: ', z)
print('P-value: ', p_value)
print('Standard Error: ', SE)
print('95% CI: ', CI)
N/2

Z-score:  -3.90156062425
P-value:  0.0
Standard Error:  0.051
95% CI:  [ 128.  173.]


196.0

## Q3 Response:

Results show that under normality assumption with 5% level of significance:

1. The the low sample mean for black calls is unlikely be explained by randomness.
2. High z-score and low p-value confirm this
3. 95% confidence interval falls indicates that true mean for black calls lies between 127 and 173 which is below the mid-point for the sample size N/2= 196.

All this leads to the conclusion that the null hypothesis can be rejected.



